### PR TITLE
NAS-109996 / 21.06 / Fixed display device issues for VMs

### PR DESCRIPTION
**Pull request by ryousaf**
NAS-109996

### CODE REVIEW

File: src/app/pages/vm/vm-list/vm-list.component.ts [hunk 1]
```diff
@@ -238,6 +238,18 @@ export class VMListComponent implements OnDestroy {
             }
         });
     }
+    extractHostname(url: string) {
```
Comment (denysbutenko) [inline src/app/pages/vm/vm-list/vm-list.component.ts:241]: what type returns this method?

File: src/app/pages/vm/vm-list/vm-list.component.ts [hunk 2]
```diff
@@ -455,60 +467,72 @@ export class VMListComponent implements OnDestroy {
...
+                            title: T("Display Device"),
+                            message: T("Pick a display device to open"),
+                            fieldConfig: [{
+                                type: 'radio',
+                                name: 'display_device',
+                                placeholder: T("Display Device"),
+                                options: display_devices_res.map((d) => {return {label: d.attributes.type, value: d.id};}),
+                                validation: [Validators.required],
+                            }],
+                            saveButtonText: "Open",
```
Comment (denysbutenko) [inline src/app/pages/vm/vm-list/vm-list.component.ts:491]: please wrap into `T()` to make it translatable

File: src/app/pages/vm/devices/device-edit/device-edit.component.ts [hunk 3]
```diff
@@ -296,6 +315,15 @@ export class DeviceEditComponent implements OnInit {
       inputType : 'password',
       validation: helptext.password_validation
     },
+    {
+      name : 'type',
+      placeholder : helptext.type_placeholder,
+      type: 'select',
+      options : [
+        {label: 'VNC', value: "VNC"},
```
Comment (denysbutenko) [inline src/app/pages/vm/devices/device-edit/device-edit.component.ts:323]: labels also should be translatable

**Comment by denysbutenko** [comment]
There are conflicting files

**Review by denysbutenko — APPROVED** [review]

**Review by sonicaj — CHANGES_REQUESTED** [review]
Can we please fix NAS-110377 in the same PR perhaps as middleware has updated it's response which means that using the same UI diff will not work now in latest nightlies ?

**Pull request closed without merging**